src/commonMain/kotlin/dev/romainguy/kotlin/math/Matrix.kt

```diff
@@ -497,6 +541,27 @@ fun rotation(axis: Float3, angle: Float): Mat4 {
             0.0f, 0.0f, 0.0f, 1.0f
     )
 }
+fun rotation(quaternion: Float4): Mat4 {
+    val n = normalize(quaternion)
+    val s = Mat4(n.x * n, n.y * n, n.z * n)
```

Comment:
Why build a `Mat4` here? You're going to create an allocation that seems entirely unnecessary.

Reply:
Fixed https://github.com/romainguy/kotlin-math/pull/33/commits/dcadac29ba02c545a288f0fa951cbb4d0b282cf3